lunatrace/bsl/backend/src/snapshot/node-package-tree.ts

```diff
@@ -108,14 +112,19 @@ export async function collectPackageGraphsFromDirectory(repoDir: string): Promis
       // This value is set to false by the library when there are zero dev dependencies
       const prodOrDevLabel = pkgTree.hasDevDependencies ? 'dev' : 'prod';
 
-      const pkgDependenciesWithGraphAndMetadata = Object.values(pkgTree.dependencies).map((pkg) => {
+      // Add the unique id to the version of the root node if requested so it has a unique key in the db.
+      if (rootUniqueId) {
+        pkgTree.version = `${pkgTree.version}-${rootUniqueId}`;
+      }
+
+      const pkgDependenciesWithGraphAndMetadata = [pkgTree, ...Object.values(pkgTree.dependencies)].map((pkg) => {
```

Comment:
```typescript
const treeWithMetadata = Object.values(pkgTree.dependencies)].map((pkg) => { ... });
const pkgDependenciesWithGraphAndMetadata = [pkgTree, ...treeWithMetadata];
```

Reply:
That's not what this does. It adds the root node to the list and maps over that.